File: src/main/java/ch/daplab/nosql/cassandra/doodle/services/PollService.java
package ch.daplab.nosql.cassandra.doodle.services;

import ch.daplab.nosql.cassandra.doodle.domains.Poll;
import ch.daplab.nosql.cassandra.doodle.domains.Subscriber;

import java.util.List;


public interface PollService {

	public Poll getPollById(String pollId);

	public List<Poll> getAllPolls();

	public Poll createPoll(Poll poll);

	public Poll addSubscriber(String pollId, Subscriber subscriber);

	public void deletePoll(String pollId);
	
}

File: src/main/kotlin/ch/daplab/nosql/cassandra/doodle/services/PollService.kt
package ch.daplab.nosql.cassandra.doodle.services

import ch.daplab.nosql.cassandra.doodle.domains.Poll
import ch.daplab.nosql.cassandra.doodle.domains.Subscriber
import ch.daplab.nosql.cassandra.doodle.domains.impl.DataPoll
import ch.daplab.nosql.cassandra.doodle.domains.impl.DataSubscriber


interface PollService {

    fun getPollById(pollId: String): Poll?

    val allDataPolls: List<Poll>

    fun createPoll(poll: Poll): Poll

    fun addSubscriber(pollId: String, subscriber: Subscriber): Poll

    fun deletePoll(pollId: String)

}
